scm-core/src/main/java/sonia/scm/web/UserAgent.java

```diff
@@ -212,6 +232,21 @@ public Builder browser(boolean browser)
       return this;
     }
 
+    /**
+     * Set to {@code true} if the {@link UserAgent} is an scm client.
+     *
+     *
+     * @param scmClient {@code true} for an scm client
+     *
+     * @return {@code this}
+     */
+    public Builder scmClient(boolean scmClient)
+    {
+      this.scmClient = scmClient;
+
+      return this;
+    }
+
```

Comment:
A thought just came to me: Normally we use builders, so that only valid objects can be created. Using this API, you can create a `UserAgent` that is both (browser and scm client) at the same time. I wonder whether two methods like `browser()` and `scmClient()` would be better. These could either be final methods, so that you have to choose. The `borwser(boolean)` should be deprecated in this case.
An other possibility would be to guard these methods so that you cannot set either one to `true`, when the other one is `true`, too.